woocommerce-gateway-ebanx/services/class-wc-ebanx-one-click.php

```diff
@@ -244,28 +244,28 @@ protected function customer_has_ebanx_required_data() {
 
 		$names = $this->gateway->names;
 
-		$_POST['ebanx_token'] = $card->token;
-		$_POST['ebanx_masked_card_number'] = $card->masked_number;
-		$_POST['ebanx_brand'] = $card->brand;
-		$_POST['ebanx_billing_cvv'] = WC_EBANX_Request::read('ebanx-one-click-cvv');
-		$_POST['ebanx_is_one_click'] = true;
-		$_POST['ebanx_billing_instalments'] = WC_EBANX_Request::read('ebanx-credit-card-installments');
-
-		$_POST[$names['ebanx_billing_brazil_document']] = get_user_meta( $this->userId, '_ebanx_billing_brazil_document', true );
-		$_POST[$names['ebanx_billing_brazil_birth_date']] = get_user_meta( $this->userId, '_ebanx_billing_brazil_birth_date', true );
-
-		$_POST['billing_postcode']  = $this->get_user_billing_address()['postcode'];
-		$_POST['billing_address_1'] = $this->get_user_billing_address()['address_1'];
-		$_POST['billing_city']      = $this->get_user_billing_address()['city'];
-		$_POST['billing_state']     = $this->get_user_billing_address()['state'];
-
-		return empty( WC_EBANX_Request::read('ebanx-one-click-token') )
-			|| empty( WC_EBANX_Request::read('ebanx-credit-card-installments') )
-			|| empty( WC_EBANX_Request::read('ebanx-one-click-cvv') )
-			|| ! WC_EBANX_Request::has($names['ebanx_billing_brazil_document'])
-			|| ! WC_EBANX_Request::has($names['ebanx_billing_brazil_birth_date'])
-			|| empty( WC_EBANX_Request::read($names['ebanx_billing_brazil_document']) )
-			|| empty( WC_EBANX_Request::read($names['ebanx_billing_brazil_birth_date']) );
+		WC_EBANX_Request::set('ebanx_token', $card->token);
+		WC_EBANX_Request::set('ebanx_masked_card_number', $card->masked_number);
+		WC_EBANX_Request::set('ebanx_brand', $card->brand);
+		WC_EBANX_Request::set('ebanx_billing_cvv', WC_EBANX_Request::read('ebanx-one-click-cvv'));
+		WC_EBANX_Request::set('ebanx_is_one_click', true);
+		WC_EBANX_Request::set('ebanx_billing_instalments', WC_EBANX_Request::read('ebanx-credit-card-installments'));
+
+		WC_EBANX_Request::set($names['ebanx_billing_brazil_document'], get_user_meta( $this->userId, '_ebanx_billing_brazil_document', true ));
+		WC_EBANX_Request::set($names['ebanx_billing_brazil_birth_date'], get_user_meta( $this->userId, '_ebanx_billing_brazil_birth_date', true ));
+
+		WC_EBANX_Request::set('billing_postcode', $this->get_user_billing_address()['postcode']);
+		WC_EBANX_Request::set('billing_address_1', $this->get_user_billing_address()['address_1']);
+		WC_EBANX_Request::set('billing_city', $this->get_user_billing_address()['city']);
+		WC_EBANX_Request::set('billing_state', $this->get_user_billing_address()['state']);
+
+		return WC_EBANX_Request::has('ebanx-one-click-token')
+			&& WC_EBANX_Request::has('ebanx-credit-card-installments')
+			&& WC_EBANX_Request::has('ebanx-one-click-cvv')
```

Comment:
`ebanx-one-click-token`, `ebanx-credit-card-installments`, `ebanx-one-click-cvv`: same, check empty.
The if should be all "||" and the has lines negated.